File: Find-Missing-and-Repeated-Values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        HashMap<Integer, Integer> mpp= new HashMap<>();
        int gridSize = grid.length;
        for(int i = 0 ; i < gridSize ; i+=1){
            for(int j = 0 ; j < gridSize ; j+=1){
                mpp.put(grid[i][j], mpp.getOrDefault(grid[i][j], 0)+1);
            }
        }
        int totalSize = gridSize * gridSize;
        int duplicate = -1;
        int missing = -1;
        for(int i = 1; i <= totalSize ; i+=1 ){
            if(!mpp.containsKey(i)){
                duplicate = i;
            }
            if(mpp.containsKey(i)&& mpp.get(i) == 2){
                missing = i;
            }
        }
        return new int[] {missing, duplicate};
    }
}

File: 3227-find-missing-and-repeated-values/find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        missing_num = -1
        repeated = -1
        seen = set()
        missing = set()
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] in seen:
                    repeated = grid[i][j]
                    
                seen.add(grid[i][j])
                missing.add(grid[i][j])
        missing = sorted(missing)
        print(missing)
        for i in range(1, len(missing)+1):
            if i != missing[i-1]:
                missing_num = i
                break
        print(missing_num)
        if missing_num == -1:
            missing_num = len(missing) + 1
        return [repeated, missing_num]

        
        

        